### PR TITLE
[pkg/stanza] container: move k8s metadata into resource attributes

### LINKED ISSUE #33341: [pkg/stanza] container parser should add k8s metadata as resource attributes
### Component(s)

pkg/stanza

### Is your feature request related to a problem? Please describe.

At the moment the newly added [`container` parser extracts some k8s metadata from the file paths](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/pkg/stanza/docs/operators/container.md#add-metadata-from-file-path).
However, these are added as `Attributes` and not as `Resource attributes`.

### Describe the solution you'd like

The k8s metadata should be added as Resource Attributes instead of Attributes.

Expected log records:
```
Resource SchemaURL: 
Resource attributes:
     -> k8s.namespace.name: Str(default)
     -> k8s.pod.name: Str(daemonset-logs-6qglg)
     -> k8s.container.restart_count: Str(0)
     -> k8s.pod.uid: Str(898553cb-ce78-4a45-bc7b-8464d56ec894)
     -> k8s.container.name: Str(busybox)
ScopeLogs #0
ScopeLogs SchemaURL: 
InstrumentationScope  
LogRecord #0
ObservedTimestamp: 2024-06-03 15:19:20.052244563 +0000 UTC
Timestamp: 2024-06-03 15:19:19.896279612 +0000 UTC
SeverityText: 
SeverityNumber: Unspecified(0)
Body: Str(otel logs at 15:19:19)
Attributes:
     -> log.file.path: Str(/var/log/pods/default_daemonset-logs-6qglg_898553cb-ce78-4a45-bc7b-8464d56ec894/busybox/0.log)
     -> time: Str(2024-06-03T15:19:19.896279612Z)
     -> logtag: Str(F)
     -> log.iostream: Str(stdout)
```

That's also aligned with the [Helm preset](https://github.com/open-telemetry/opentelemetry-helm-charts/blob/55496062d22470310f90f011e5cfec1e4b08ceba/charts/opentelemetry-collector/templates/_config.tpl#L264-L278).

### Describe alternatives you've considered

_No response_

### Additional context

Would that be considered as a breaking change though since the parser has already been released?
On the other hand since the main goal of the parser is to replicate the existent behavior of the [Helm preset](https://github.com/open-telemetry/opentelemetry-helm-charts/blob/55496062d22470310f90f011e5cfec1e4b08ceba/charts/opentelemetry-collector/templates/_config.tpl#L264-L278) I would consider it as a bug fix for something that we need to fix anyways.

### PR DESCRIPTION
**Pull request by ChrsMark**
**Description:** <Describe what has changed.>

Container parser should add k8s metadata as resource attributes and not as log record attributes.

**Link to tracking Issue:** <Issue number if applicable> Fixes https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/33341

**Testing:** <Describe what testing was performed and which tests were added.>
Manual testing on local k8s cluster:

```console
2024-06-04T06:40:08.219Z	info	ResourceLog #0
Resource SchemaURL: 
Resource attributes:
     -> k8s.pod.uid: Str(d5ecc924-e255-4525-b5be-6437939b1e4d)
     -> k8s.container.name: Str(busybox)
     -> k8s.namespace.name: Str(default)
     -> k8s.pod.name: Str(daemonset-logs-dhzcq)
     -> k8s.container.restart_count: Str(0)
ScopeLogs #0
ScopeLogs SchemaURL: 
InstrumentationScope  
LogRecord #0
ObservedTimestamp: 2024-06-04 06:40:08.007370503 +0000 UTC
Timestamp: 2024-06-04 06:40:07.855932421 +0000 UTC
SeverityText: 
SeverityNumber: Unspecified(0)
Body: Str(otel logs at 06:40:07)
Attributes:
     -> logtag: Str(F)
     -> key2: Map({"key_in":"val2"})
     -> log.file.path: Str(/var/log/pods/default_daemonset-logs-dhzcq_d5ecc924-e255-4525-b5be-6437939b1e4d/busybox/0.log)
     -> time: Str(2024-06-04T06:40:07.855932421Z)
     -> log.iostream: Str(stdout)
Trace ID: 
Span ID: 
Flags: 0
LogRecord #1
ObservedTimestamp: 2024-06-04 06:40:08.007451031 +0000 UTC
Timestamp: 2024-06-04 06:40:07.957875321 +0000 UTC
SeverityText: 
SeverityNumber: Unspecified(0)
Body: Str(otel logs at 06:40:07)
Attributes:
     -> log.file.path: Str(/var/log/pods/default_daemonset-logs-dhzcq_d5ecc924-e255-4525-b5be-6437939b1e4d/busybox/0.log)
     -> log.iostream: Str(stdout)
     -> time: Str(2024-06-04T06:40:07.957875321Z)
     -> key2: Map({"key_in":"val2"})
     -> logtag: Str(F)
Trace ID: 
Span ID: 
Flags: 0
```

**Documentation:** <Describe the documentation added.> ~

### CODE REVIEW

File: pkg/stanza/docs/operators/container.md [hunk 1]
```diff
@@ -31,16 +31,18 @@ will produce the following k8s metadata:
...
+      "k8s": {
+        "container": {
+          "name": "kube-controller",
+          "restart_count": "1"
+        }, "pod": {
+          "uid": "49cc7c1fd3702c40b2686ea7486091d6",
+          "name": "kube-controller-kind-control-plane"
+        }, "namespace": {
+          "name": "some-ns"
+        }
```
Comment (djaglowski) [inline pkg/stanza/docs/operators/container.md:45]: Didn't we flatten these?

File: pkg/stanza/operator/parser/container/parser_test.go [hunk 2]
```diff
@@ -157,16 +157,18 @@ func TestProcess(t *testing.T) {
 			},
 			&entry.Entry{
 				Attributes: map[string]any{
-					"time":                        "2029-03-30T08:31:20.545192187Z",
-					"log.iostream":                "stdout",
+					"time":          "2029-03-30T08:31:20.545192187Z",
```
Comment (djaglowski) [inline pkg/stanza/operator/parser/container/parser_test.go:160]: I know it's not from this PR, but any reason to keep original timestamp after we parse it?

File: pkg/stanza/docs/operators/container.md (hunk 1 above)
Comment (ChrsMark) [reply pkg/stanza/docs/operators/container.md]: I "wrongly" fixed a wrong left-over. Thank's!

File: pkg/stanza/operator/parser/container/parser_test.go (hunk 2 above)
Comment (ChrsMark) [reply pkg/stanza/operator/parser/container/parser_test.go:160]: I think the original approach was preserving the original time after parsing. The `time_parser` does not remove the original time field from what I can see in its [docs](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/pkg/stanza/docs/types/timestamp.md#parse-a-timestamp-using-a-strptime-layout).
So the question is if we want to follow the same approach here for compatibility reasons for those users that might still use these for any reason.
Based on the above, leaving it as is and discussing (in follow-up issue) its removal through an optional setting might be the safest path?

Comment (djaglowski) [reply pkg/stanza/operator/parser/container/parser_test.go:160]: Makes sense. I am in favor of removing it after parsing, but a followup PR makes sense.

**Review by djaglowski — APPROVED** [review]

**Pull request merged**